### DISCUSSION TITLE
[help] how to paste the selected string onto terminal and not execute the command

### DISCUSSION THREAD

**Discussion by zhuzhzh** [junegunn/fzf#3081] [Q&A]
- [x ] I have read through the manual page (`man fzf`)
- [x] I have the latest version of fzf
- [x] I have searched through the existing issues

## Info

- OS
    - [x] Linux
    - [ ] Mac OS X
    - [ ] Windows
    - [ ] Etc.
- Shell
    - [ ] bash
    - [x] zsh
    - [ ] fish

## Problem / Steps to reproduce

I want to use fzf and zoxide as path selection for any command.

for ex:

```
cp $(zoxide query --list |fzf --print0)<CR>
```

when I selected the path in fzf TUI,  the command `cp` will execute and report the destinamtion is missing.
I hope the selected path is pasted on the place only.

I am not familar with zsh zle. I tried to emulate the built-in fzf script and use `alt+o` to trigger completion. but it doesn't work.
```
  zoxide-history-widget-accept() {
    local selected
    selected=( $(zoxide query --list |fzf --print0) )
    local ret=$?
    echo -n "$selected"
    return $ret
  }
  zle    -N    zoxide-history-widget-accept
  bindkey '^[o' zoxide-history-widget-accept
```

## fzf solution

following `Custom fuzzy completion`

I use the following method.

```zsh
  _fzf_complete_cp() {
    _fzf_complete +m -- "$@" < <(
      /home/harriszh/.cargo/bin/zoxide query --list
    )
  }
## used to strip the tailing space
  _fzf_complete_cp_post() {
    xargs echo -n
  }
```

then I input  `cp **<TBA>`, it will paste the selection on the command.  I want to know if it's possible to paste the selected string got from fzf into the command line instead of defining completion for the comand one by one.

**Comment by ntevenhere** [comment]
I think you want `--bind tab:replace-query`.
Mind that I only read the title of your post.

**Comment by LangLangBart** [comment]
- `print -z` maybe ?
- see:
  -  zsh guide - [yamadapc/zshguide builtins-for-printing](https://github.com/yamadapc/zshguide/blob/master/chapter3.md#321-builtins-for-printing) or
  -  [zsh-manual](https://zsh.sourceforge.io/Doc/Release/Shell-Builtin-Commands.html#Shell-Builtin-Commands) or 
  - to get the documentation of the builtin into the terminal type `print` press `Alt+Shift+h`.

- example 
```sh
# select script command from package.json and place it on the command line, but don't execute it
nr() {
	local script
	script=$(jq -r '.scripts | to_entries[] | [.key, .value] | @tsv' <package.json | sort |
		fzf --exit-0 --delimiter '\t' --with-nth 1 --height 45% \
			--cycle --border-label " package.json script " \
			--border rounded --color 'border:255,label:244:bold' \
			--preview-window "nohidden:down:border:wrap" \
			--preview 'bat -pl sh <<<{2}')
	print -nz "pnpm --silent run ${script%%$'\t'*} $*"
}
```
